conn_test.go

```diff
@@ -797,3 +797,26 @@ func TestFailedConnectionReadPanic(t *testing.T) {
 	}
 	t.Fatal("should not get here")
 }
+
+type fakeNetClosedReader struct {
+}
+
+func (r fakeNetClosedReader) Read([]byte) (int, error) {
+	return 0, net.ErrClosed
+}
+
+func TestConnectionClosed(t *testing.T) {
+	t.Parallel()
```

Comment:
I'm not sure if we really need this test to be run in Parallel?

Reply:
Removed t.Parallel

Reply:
I could go either way but I'm in favor of leaving the parallel test for consistency since https://github.com/gorilla/websocket/pull/872 was recently merged.

Reply:
Lets leave it as is and not require more changes.